components/service/sync-autofill/src/main/java/mozilla/components/service/sync/autofill/AutofillCreditCardsAddressesStorage.kt

```diff
@@ -24,34 +30,96 @@ const val AUTOFILL_DB_NAME = "autofill.sqlite"
  * library.
  */
 class AutofillCreditCardsAddressesStorage(
-    context: Context
-) : CreditCardsAddressesStorage, AutoCloseable {
+    context: Context,
+    securePrefs: Lazy<SecureAbove22Preferences>
+) : CreditCardsAddressesStorage, SyncableStore, AutoCloseable {
+    private val logger = Logger("AutofillCCAddressesStorage")
 
     private val coroutineContext by lazy { Dispatchers.IO }
 
+    val crypto by lazy { AutofillCrypto(context, securePrefs.value) }
+
     private val conn by lazy {
         AutofillStorageConnection.init(dbPath = context.getDatabasePath(AUTOFILL_DB_NAME).absolutePath)
         AutofillStorageConnection
     }
 
-    override suspend fun addCreditCard(creditCardFields: UpdatableCreditCardFields): CreditCard =
-        withContext(coroutineContext) {
-            conn.getStorage().addCreditCard(creditCardFields.into()).into()
+    private fun newKeyHandler(reason: KeyGenerationReason) {
+        when (reason) {
+            KeyGenerationReason.New -> logger.info("CC key generated")
```

Comment:
You could argue that the AS recovery be done for all cases here - `.New` implies storage will be empty anyway and if you somehow managed to get here when it was not new you'd end up in pain.